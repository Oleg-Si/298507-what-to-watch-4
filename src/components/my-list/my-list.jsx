import React from 'react';
import PropTypes from 'prop-types';
import AppHeaderMyList from './../app-header-my-list/app-header-my-list.jsx';
import AppFooter from './../app-footer/app-footer.jsx';
import {connect} from 'react-redux';
import dataOperations from './../../redux/data/operations';
import {getFavoriteFilms} from '../../redux/data/selectors.js';
import FilmList from './../film-list/film-list.jsx';
import {getAuthorizationStatus, getUserAvatar} from './../../redux/user/selectors';


const MyList = (props) => {
  const {
    films,
    onFilmCardTitleClick,
    authorizationStatus,
    userAvatar
  } = props;

  return (
    <div className="user-page">

      <AppHeaderMyList
        authorizationStatus={authorizationStatus}
        userAvatar={userAvatar}
      >
        <h1 className="page-title user-page__title">My list</h1>
      </AppHeaderMyList>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <FilmList
          films={films}
          filmsCount={films.length}
          onFilmCardTitleClick={onFilmCardTitleClick}
        />
      </section>

      <AppFooter />
    </div>
  );
};

MyList.propTypes = {
  films: PropTypes.array.isRequired,
  onFilmCardTitleClick: PropTypes.func.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  userAvatar: PropTypes.string,
};

const mapStateToProps = (state) => ({
  films: getFavoriteFilms(state),
  authorizationStatus: getAuthorizationStatus(state),
  userAvatar: getUserAvatar(state)
});

const mapDispatchToProps = (dispatch) => ({
  onFilmCardTitleClick(film) {
    dispatch(dataOperations.loadComments(film.id));
  }
});


export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
