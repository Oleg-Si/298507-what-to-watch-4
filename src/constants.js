export const DEFAULT_GENRE = `All genres`;
export const DEFAULT_TAB = `Overview`;
export const FILM_COUNT = 40;
export const RENDERED_FILMS_COUNT = 8;
export const COUNT_MORE_FILMS = 8;
export const COUNT_MORE_LIKE_THIS_FILMS = 4;

export const Tabs = [
  `Overview`,
  `Details`,
  `Reviews`
];

export const RatingKey = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};

export const Screens = {
  MAIN: `Main`,
  FILM_PAGE: `FilmPage`
};
