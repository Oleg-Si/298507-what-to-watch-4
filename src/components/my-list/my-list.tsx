import * as React from 'react';
import AppHeaderMyList from '../app-header-my-list/app-header-my-list';
import AppFooter from '../app-footer/app-footer';
import {connect} from 'react-redux';
import dataOperations from '../../redux/data/operations';
import {getFavoriteFilms} from '../../redux/data/selectors';
import FilmList from '../film-list/film-list';
import {getAuthorizationStatus, getUserAvatar} from '../../redux/user/selectors';
import {AuthorizationStatus} from '../../constants';
import {FilmInterface} from './../../types';

interface Props {
  films: FilmInterface[],
  onFilmCardTitleClick: (film: FilmInterface) => void,
  onMyListClick: () => void,
  authorizationStatus: AuthorizationStatus,
  userAvatar: string,
}

const MyList: React.FC<Props> = (props: Props) => {
  const {
    films,
    onFilmCardTitleClick,
    authorizationStatus,
    userAvatar,
    onMyListClick
  } = props;

  return (
    <div className="user-page">

      <AppHeaderMyList
        authorizationStatus={authorizationStatus}
        userAvatar={userAvatar}
        onMyListClick={onMyListClick}
      >
        <h1 className="page-title user-page__title">My list</h1>
      </AppHeaderMyList>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <FilmList
          films={films}
          filmsCount={films.length}
          onFilmCardTitleClick={onFilmCardTitleClick}
        />
      </section>

      <AppFooter />
    </div>
  );
};

const mapStateToProps = (state) => ({
  films: getFavoriteFilms(state),
  authorizationStatus: getAuthorizationStatus(state),
  userAvatar: getUserAvatar(state)
});

const mapDispatchToProps = (dispatch) => ({
  onFilmCardTitleClick(film) {
    dispatch(dataOperations.loadComments(film.id));
  },

  onMyListClick() {
    dispatch(dataOperations.loadFavoriteFilms());
  }
});


export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
