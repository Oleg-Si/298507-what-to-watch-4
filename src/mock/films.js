import {getRandomInt, getRandomArrayItems} from './../utils';
import {FILM_COUNT} from '../constants';

const getFilms = (count) => {
  const allFilms = new Array(count).fill(``);

  allFilms.forEach((el, i, arr) => {
    const element = getMockFilm();
    element.id = i;
    arr[i] = element;
  });

  return allFilms;
};

const getMockFilm = () => ({
  id: 0,
  title: [
    `Fantastic Beasts: The Crimes of Grindelwald`,
    `Bohemian Rhapsody`,
    `Macbeth`,
    `Aviator`,
    `We need to talk about Kevin`,
    `What We Do in the Shadows`,
    `Revenant`,
    `Johnny English`,
    `Shutter Island`,
    `Pulp Fiction`,
    `No Country for Old Men`,
    `Snatch`
  ][Math.floor(Math.random() * 12)],
  img: [
    `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    `img/bohemian-rhapsody.jpg`,
    `img/macbeth.jpg`,
    `img/aviator.jpg`,
    `img/we-need-to-talk-about-kevin.jpg`,
    `img/what-we-do-in-the-shadows.jpg`,
    `img/revenant.jpg`,
    `img/johnny-english.jpg`,
    `img/shutter-island.jpg`,
    `img/pulp-fiction.jpg`,
    `img/no-country-for-old-men.jpg`,
    `img/snatch.jpg`
  ][Math.floor(Math.random() * 12)],
  poster: ``,
  genre: [
    `Dramas`,
    `Horror`,
    `Comedies`,
    `Romance`,
    `Crime`,
    `Sci-Fi`,
    `Documentary`,
    `Thrillers`,
    `Kids & Family`
  ][getRandomInt(0, 6)],
  src: getRandomArrayItems([`https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4`, `https://interactive-examples.mdn.mozilla.net/media/examples/flower.mp4`, `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`, `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`, `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`], 1),
  rating: `${getRandomInt(0, 9)}.${getRandomInt(0, 9)}`,
  releaseDate: getRandomInt(1262293200000, 1552915226386),
  runTime: getRandomInt(4000, 7800),
  ratingCount: getRandomInt(5, 250),
  description: getRandomArrayItems([`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`, `Gustave prides himself on providing first-class service to the hotel&apos;s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave&aposs lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`, `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`, `Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante. Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`, `Sed sed nisi sed augue convallis suscipit in sed felis. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`], getRandomInt(2, 5)),
  director: getRandomArrayItems([
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ], 1),
  starring: getRandomArrayItems([
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ], getRandomInt(3, 12)),
  reviews: getMockReviews(getRandomInt(3, 7))
});

const getMockReviews = (count) => new Array(count).fill(``).map(() => getReviews());

const getReviews = () => ({
  author: getRandomArrayItems([
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ], 1),
  rating: `${getRandomInt(0, 9)}.${getRandomInt(0, 9)}`,
  date: getRandomInt(1262293200000, 1552915226386),
  comment: getRandomArrayItems([
    `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
  ], 1)
});

const promoFilmMock = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`
};

const mockFilmForTests = {
  id: 0,
  title: `Revenant`,
  img: `img/revenant.jpg`,
  genre: `Dramas`,
  rating: `8.5`,
  src: `https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4`,
  releaseDate: 1552915226386,
  runTime: 5084,
  ratingCount: 148,
  description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`, `Gustave prides himself on providing first-class service to the hotel&apos;s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave&aposs lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  director: `Bill Murray`,
  starring: [`Edward Norton`, `Jude Law`, `Willem Dafoe and other`, `Wes Andreson`],
  reviews: [
    {
      author: `Edward Norton`,
      rating: `3.5`,
      date: 1500713355882,
      comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
    },
    {
      author: `Willem Dafoe`,
      rating: `6.7`,
      date: 1302290042303,
      comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
    },
    {
      author: `Ralph Fiennes`,
      rating: `8.9`,
      date: 1445604462862,
      comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
    }
  ]
};

const mockFilmsForTests = [
  {
    id: 0,
    title: `Aviator`,
    img: `img/aviator.jpg`,
    genre: `Comedies`,
    rating: `8.1`,
    src: `https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4`,
    releaseDate: 1552905226386,
    runTime: 6104,
    ratingCount: 198,
    description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`],
    director: `Bill Murray`,
    starring: [`Edward Norton`, `Jude Law`, `Willem Dafoe and other`, `Wes Andreson`],
    reviews: [
      {
        author: `Edward Norton`,
        rating: `3.5`,
        date: 1500793355882,
        comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
      },
      {
        author: `Willem Dafoe`,
        rating: `6.7`,
        date: 1302908542303,
        comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
      },
      {
        author: `Ralph Fiennes`,
        rating: `8.9`,
        date: 1444604462862,
        comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
      }
    ]
  },
  {
    id: 1,
    title: `Macbeth`,
    img: `img/macbeth.jpg`,
    genre: `Horror`,
    rating: `3.6`,
    src: `https://interactive-examples.mdn.mozilla.net/media/examples/flower.mp4`,
    releaseDate: 1552005226386,
    ratingCount: 248,
    description: [`Gustave prides himself on providing first-class service to the hotel&apos;s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave&aposs lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Edward Norton`,
    starring: [`Jude Law`, `Willem Dafoe and other`, `Wes Andreson`],
    reviews: [
      {
        author: `Edward Norton`,
        rating: `3.5`,
        date: 1500793355882,
        comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
      },
      {
        author: `Willem Dafoe`,
        rating: `6.7`,
        date: 1302908542303,
        comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
      },
      {
        author: `Ralph Fiennes`,
        rating: `8.9`,
        date: 1444604462862,
        comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
      }
    ]
  },
  {
    id: 2,
    title: `Snatch`,
    img: `img/snatch.jpg`,
    genre: `Documentary`,
    rating: `2.5`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    releaseDate: 1552518826386,
    runTime: 7100,
    ratingCount: 850,
    description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`, `Gustave prides himself on providing first-class, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Jude Law`,
    starring: [`Edward Norton`, `Willem Dafoe and other`, `Wes Andreson`],
    reviews: [
      {
        author: `Edward Norton`,
        rating: `3.5`,
        date: 1500793355882,
        comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
      },
      {
        author: `Willem Dafoe`,
        rating: `6.7`,
        date: 1302908542303,
        comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
      },
      {
        author: `Ralph Fiennes`,
        rating: `8.9`,
        date: 1444604462862,
        comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
      }
    ]
  }
];

const films = getFilms(FILM_COUNT);

export default films;
export {promoFilmMock, mockFilmForTests, mockFilmsForTests};
