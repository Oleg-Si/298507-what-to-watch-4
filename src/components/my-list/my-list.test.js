import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import history from './../../history';
import {AuthorizationStatus} from './../../constants';
import {MyList} from './my-list.jsx';
import {mockFilmsForTests} from '../../mock/films';

it(`Проверяет снепшот компонента MyList`, () => {
  const tree = renderer.create(
      <Router
        history={history}
      >
        <MyList
          films={mockFilmsForTests}
          onFilmCardTitleClick={() => {}}
          onMyListClick={() => {}}
          authorizationStatus={AuthorizationStatus.AUTH}
          userAvatar={`https://4.react.pages.academy/wtw/static/avatar/2.jpg`}
        />
      </Router>, {
        createNodeMock: () => {
          return {};
        }}
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
