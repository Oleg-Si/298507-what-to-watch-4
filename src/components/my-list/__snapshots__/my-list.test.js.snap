// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяет снепшот компонента MyList 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="https://4.react.pages.academy/wtw/static/avatar/2.jpg"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="preloader"
          style={
            Object {
              "left": "50%",
              "marginLeft": "-32px",
              "marginTop": "-32px",
              "position": "absolute",
              "top": "50%",
              "zIndex": "2",
            }
          }
        >
          <svg
            height="64px"
            viewBox="0 0 128 128"
            width="64px"
            xmlSpace="preserve"
          >
            <rect
              fill="#FFFFFF"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <g>
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="16"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="14.344"
                transform="rotate(45 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="12.531"
                transform="rotate(90 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="10.75"
                transform="rotate(135 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="10.063"
                transform="rotate(180 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="8.063"
                transform="rotate(225 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="6.438"
                transform="rotate(270 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="5.375"
                transform="rotate(315 64 64)"
              />
              <animateTransform
                attributeName="transform"
                calcMode="discrete"
                dur="720ms"
                repeatCount="indefinite"
                type="rotate"
                values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
              />
            </g>
          </svg>
        </div>
        <div
          className="small-movie-card__image"
        >
          <video
            height={175}
            poster="img/aviator.jpg"
            src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/0"
            onClick={[Function]}
          >
            Aviator
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="preloader"
          style={
            Object {
              "left": "50%",
              "marginLeft": "-32px",
              "marginTop": "-32px",
              "position": "absolute",
              "top": "50%",
              "zIndex": "2",
            }
          }
        >
          <svg
            height="64px"
            viewBox="0 0 128 128"
            width="64px"
            xmlSpace="preserve"
          >
            <rect
              fill="#FFFFFF"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <g>
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="16"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="14.344"
                transform="rotate(45 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="12.531"
                transform="rotate(90 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="10.75"
                transform="rotate(135 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="10.063"
                transform="rotate(180 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="8.063"
                transform="rotate(225 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="6.438"
                transform="rotate(270 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="5.375"
                transform="rotate(315 64 64)"
              />
              <animateTransform
                attributeName="transform"
                calcMode="discrete"
                dur="720ms"
                repeatCount="indefinite"
                type="rotate"
                values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
              />
            </g>
          </svg>
        </div>
        <div
          className="small-movie-card__image"
        >
          <video
            height={175}
            poster="img/macbeth.jpg"
            src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="preloader"
          style={
            Object {
              "left": "50%",
              "marginLeft": "-32px",
              "marginTop": "-32px",
              "position": "absolute",
              "top": "50%",
              "zIndex": "2",
            }
          }
        >
          <svg
            height="64px"
            viewBox="0 0 128 128"
            width="64px"
            xmlSpace="preserve"
          >
            <rect
              fill="#FFFFFF"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <g>
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="16"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="14.344"
                transform="rotate(45 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="12.531"
                transform="rotate(90 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="10.75"
                transform="rotate(135 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="10.063"
                transform="rotate(180 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="8.063"
                transform="rotate(225 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="6.438"
                transform="rotate(270 64 64)"
              />
              <circle
                cx="16"
                cy="64"
                fill="#000000"
                fillOpacity="1"
                r="5.375"
                transform="rotate(315 64 64)"
              />
              <animateTransform
                attributeName="transform"
                calcMode="discrete"
                dur="720ms"
                repeatCount="indefinite"
                type="rotate"
                values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
              />
            </g>
          </svg>
        </div>
        <div
          className="small-movie-card__image"
        >
          <video
            height={175}
            poster="img/snatch.jpg"
            src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            Snatch
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2020 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
